src/components/widgets/DateWidget.js

```diff
@@ -22,9 +12,9 @@ function DateWidget({
     <input type="date"
       id={id}
       className="form-control"
-      value={fromJSONDate(value)}
+      value={typeof value === "undefined" ? "" : value}
       required={required}
-      onChange={(event) => onChange(toJSONDate(event.target.value))} />
+      onChange={(event) => onChange(event.target.value)} />
```

Comment:
Because native HTML5 `input[type=date]` no more outputs JSON datetimes but string formatted as `YYYY-MM-DD`, which is exactly what we want (we doesn't want datetimes for dates anymore).